bin/jruby.sh

```diff
@@ -805,6 +876,15 @@ if $print_environment_log; then
     exit 0
 fi
 
+# ----- Perform final mutations after logging ---------------------------------
+if $remove_jsa_files; then
+    # Delete All AppCDS files if requested
+    eval rm -f -- "$jruby_jsa_files"
+elif $regenerate_jsa_file && [ -e "$jruby_jsa_file" ]; then
+    # Delete AppCDS file if requested or if it's outdated
+    rm -f -- "$jruby_jsa_file"
+fi
```

Comment:
I'm not sure this section is doing what we want:

* My thought was that --rmcache would run nothing but clear those files. As it is now if you don't pass a Ruby script or `-e` it will wait on stdin for code.
* Doesn't this `$regenerate_jsa_file` line cause it to immediately wipe out the archive it just regenerated? I thought it would but can't reproduce thiat with `jruby --cache -e 1`

Reply:
- Oh yeah it doesn't make sense to proceed there if you can already use the `--cache` option
- No, the launcher doesn't touch the archive except for here. That's why I'm checking if it exists first, because otherwise we don't need to delete it and it'll get generated by Java